app/uk/gov/hmrc/tai/repositories/CodingComponentRepository.scala

```diff
@@ -34,4 +34,9 @@ class CodingComponentRepository @Inject()(taxAccountRepository: TaxAccountReposi
   def codingComponents(nino: Nino, year: TaxYear)(implicit hc: HeaderCarrier): Future[Seq[CodingComponent]] =
     taxAccountRepository.taxAccount(nino, year)
       .map(_.as[Seq[CodingComponent]](codingComponentReads))
+
+  def codingComponentsForTaxCodeId(nino: Nino, taxCodeId: Int)(implicit hc: HeaderCarrier): Future[Seq[CodingComponent]] = {
```

Comment:
Could change the method name and parameter name here too or would this be a different bounded context :)